### PR TITLE
[datadog exporter] Remove option to change the namespace prefix

### PR DESCRIPTION
**Pull request by albertvaka**
**Description:**
Removes the config option "namespace" and hardcodes "otel" instead, which
is what the Datadog backend expects.

Additionally, stops treating the 'running' metric as a special case that
didn't use the configured namespace and makes it be processed like a
regular metric.

**Testing:** 

Updated tests.

### CODE REVIEW

**Review by KSerrania — APPROVED** [review]
Approving on behalf of Datadog

File: exporter/datadogexporter/metrics_exporter_test.go
Comment (mx-psi) [inline exporter/datadogexporter/metrics_exporter_test.go:92]: nit: we can move this test to the metrics package

**Review by mx-psi — APPROVED** [review]

File: exporter/datadogexporter/metrics/utils.go
Comment (jpkrohling) [inline exporter/datadogexporter/metrics/utils.go:50]: Probably not for this PR, but wouldn't it make sense to align with other components and use OC for the metrics? This way, people would have a single place to check metrics about all exporters, instead of "all exporters except datadog". Example: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/master/processor/groupbytraceprocessor/metrics.go

File: exporter/datadogexporter/config/config.go
Comment (jpkrohling) [inline exporter/datadogexporter/config/config.go:59]: What happens with people already using this? I know we are still allowed to make breaking changes, but establishing a procedure for handling this kind of scenario is what 0.x is for as well ;-)

Comment (mx-psi) [reply exporter/datadogexporter/config/config.go:59]: This component has not yet been in a 0.x release (it should be on the next one), I think a breaking change of this kind on a nightly is acceptable

File: exporter/datadogexporter/metrics/utils.go
Comment (mx-psi) [reply exporter/datadogexporter/metrics/utils.go:50]: Not sure if I understand your point, those metrics are there for observability of the Collector itself, right? This default metric added in #1354 is reported directly to the backend (see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1354#issuecomment-716506278 for an explanation about the usage), it is not something that you can/want to check about the component. Maybe we should clarify that in the function docs?

Comment (jpkrohling) [reply exporter/datadogexporter/metrics/utils.go:50]: Yes, sorry, I thought this particular one was related to inferring the state of the exporter, which would make it suitable to be placed along with the other exporter's metrics.

File: exporter/datadogexporter/config/config.go
Comment (jpkrohling) [reply exporter/datadogexporter/config/config.go:59]: You can keep the field in the API (config, in this case), documented as "deprecated", logging a warn in case it's used. You don't need to use its value if you are just dropping the property entirely. On the next possible release (or n+2, preferably) , you remove the field. This way, people have a chance of getting notified without reading the release notes and have a chance of scheduling this update on their end.

File: exporter/datadogexporter/metrics/utils.go
Comment (mx-psi) [reply exporter/datadogexporter/metrics/utils.go:50]: That makes sense, thanks for clarifying!

File: exporter/datadogexporter/config/config.go
Comment (mx-psi) [reply exporter/datadogexporter/config/config.go:59]: Makes sense, thank you. As I said, since this was not enabled until fairly recently I think it's safe to remove it without deprecation (so long as this PR gets merged before 0.14), but we will take that procedure into account for future breaking changes

**Review by jpkrohling — APPROVED** [review]
LGTM. I'll let a maintainer decide on the code cov diff :-)

**Pull request merged**
